server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2531,4 +2530,16 @@ void updateRefreshedCheckpoint(long checkpoint) {
             assert refreshedCheckpoint.get() >= checkpoint : refreshedCheckpoint.get() + " < " + checkpoint;
         }
     }
+
+    @Override
+    public long getMaxAutoIdTimestamp() {
```

Comment:
maybe just push the `maxUnsafeAutoIdTimestamp` up to engine and make the methods final